src/System.Management.Automation/engine/ExecutionContext.cs

```diff
@@ -1323,82 +1323,81 @@ internal void RemoveAssembly(string name)
+                {
+                    error = securityException;
+                    return null;
+                }
+            }
+
+            string fixedName = null;
+            if (!String.IsNullOrEmpty(name))
+            {
+                // Remove the '.dll' if it's there...
+                fixedName = name.EndsWith(".dll", StringComparison.OrdinalIgnoreCase)
+                                ? Path.GetFileNameWithoutExtension(name)
+                                : name;
+
+                var assemblyString = Utils.IsPowerShellAssembly(fixedName)
+                                         ? Utils.GetPowerShellAssemblyStrongName(fixedName)
+                                         : fixedName;
+
+                try
+                {
+                    loadedAssembly = Assembly.Load(new AssemblyName(assemblyString));
+                }
+                catch (FileNotFoundException fileNotFound)
+                {
+                    error = fileNotFound;
+                }
+                catch (FileLoadException fileLoadException)
+                {
+                    error = fileLoadException;
+                    // this is a legitimate error on CoreCLR for a newly emited with Add-Type assemblies
+                    // they cannot be loaded by name, but we are only interested in importing them by path
+                }
+                catch (BadImageFormatException badImage)
+                {
+                    error = badImage;
+                    return null;
+                }
+                catch (SecurityException securityException)
+                {
+                    error = securityException;
+                    return null;
+                }
+            }
+
+            if (loadedAssembly != null)
                 return loadedAssembly;
 
-            if (!String.IsNullOrEmpty(filename))
-            {
-                error = null;
-
-                try
-                {
-                    loadedAssembly = Assembly.LoadFrom(filename);
-                    return loadedAssembly;
-                }
-                catch (FileNotFoundException fileNotFound)
-                {
-                    error = fileNotFound;
-                }
-                catch (FileLoadException fileLoadException)
-                {
-                    error = fileLoadException;
-                    return null;
-                }
-                catch (BadImageFormatException badImage)
-                {
-                    error = badImage;
-                    return null;
-                }
-                catch (SecurityException securityException)
-                {
-                    error = securityException;
-                    return null;
-                }
-            }
-
 #if !CORECLR// Assembly.LoadWithPartialName is not in CoreCLR. In CoreCLR, 'LoadWithPartialName' can be replaced by Assembly.Load with the help of AssemblyLoadContext.
```

Comment:
We are removing FullCLR code, so you should remove this since you're touching this file